### ISSUE TITLE
FEATURE: Adding attributeHints to ActiveRecord

### ISSUE THREAD

**Issue by kartik-v** [yiisoft/yii2#1468]
This is more a new feature suggestion to improve for Yii 2.0. Since we now have ```hint``` functionality inbuilt as a part of Yii ```ActiveField```, it may be useful if this method is available prebuilt within a ```model```. We need to have an option within```inputOptions``` of ```ActiveField``` to default hints ( ```fieldConfig``` parameter of ```ActiveForm```).

***For example:***
 ```php
// Call the autoHint option
$form = ActiveForm::begin(['fieldConfig' => ['autoHint' => true]);
echo $form->field($model, $attribute);

// This will be equivalent to writing something like
$form->field($model, $attribute)->hint($model->getAttributeHint($attribute));
```
**Advantages**
1.  Even if we do not display this data as a hint below a form field... these hints maybe very useful to display as a message or alert at various places in our code. Most important, it allows one *single place* to manage this i.e. within the model. Its more correct to be there - as it is actually related to each model attribute.
2. One can conditionally control this data (or use scenarios) within the model.

**Feature Suggestions**

1. Add the ```attributeHints``` function to ActiveRecord.

2. Currently ```gii``` has an option of auto-creating ```attributeLabels``` from mysql table/column comments. I suggest it should actually generate the ```attributeHints``` from mysql table column comments. ```attributeLabels``` can be generated using ```camel``` conversion and other string functions available in Yii ```Inflector```.

3. The ```attributeHints``` will return an array output similar to ```attributeLabels```. 

4. We can also probably initialize default hints based on ```attributeLabels``` like below.
```php
    /**
     * Hints/Placeholders for attributes
     */
    public function attributeHints() {
        $labels = $this->attributeLabels();
        $hints = [];
        if (count($labels) > 0) {
            foreach ($labels as $attribute => $label) {
                $hints[$attribute] = Yii::t('app', 'Enter ' . strtolower($label));
            }
        }
        return $hints;
    }
```

5. Add a ```getAttributeHint``` function

```php
    /**
     * Returns the text hint for the specified attribute.
     * @param string $attribute the attribute name
     * @return string the attribute hint
     * @see attributeHints
     */
    public function getAttributeHint($attribute) {
        $hints = $this->attributeHints();
        return isset($hints[$attribute]) ? $hints[$attribute] : $this->generateAttributeLabel($attribute);
    }
```

**Comment by Faryshta** [comment]
All the other parts of the `ActiveField` rendering are handled by special methods in the `BaseHtml` class.

https://github.com/yiisoft/yii2/blob/master/framework/widgets/ActiveField.php#L180

Should we create a method `BaseHtml::activeHint($model, $attribute, $options)` ?

**Comment by samdark** [comment]
I don't think so because hint is an attribute of the input, not something that's separate from it.

**Comment by Faryshta** [comment]
@samdark the implementation on `ActiveField` is that a hint is indeed separate from an input https://github.com/yiisoft/yii2/blob/master/framework/widgets/ActiveField.php#L309

I think you mean treating the hint like a placeholder which can be done by editing the `BaseHtml::renderInput()` method

**Comment by klimov-paul** [comment]
Resolved by #8113

**Comment by kartik-v** [comment]
:+1:
